src/utils/PolygonConnector.h

```diff
@@ -81,6 +81,17 @@ class PolygonConnector
         ClosestPolygonPoint from; //!< from location in the source polygon
         ClosestPolygonPoint to; //!< to location in the destination polygon
 
+        /* 
+         * \warning Doesn't properly initialize the members
+         */
+        PolygonConnection()
+        {}
```

Comment:
In `getSecondConnection`, there is a `PolygonConnection best` instance that seems to continue existing uninitialised. The `best` variable gets initialised when there is a connection with length smaller than `best_total_distance2` (and any length will be smaller than that). But there are a lot of other conditions: `from` and `to` need to exist, it seems to break early after choosing `to_b`, and they need to be on the same side as the first connection.

The best solution to this would be to remove this constructor and force yourself to initialise it properly with known coordinates, but that's not always possible. Second-best would be to refactor/document `getSecondConnection` such that it's clear that the variable is always initialised. Third-best option would be to initialise the points to 0 here to ensure that if we get mistakes, it's at least deterministic.

Reply:
Uhm yeah okay. Option 1 please! ;)

(done)